src/libraries/System.Text.RegularExpressions/gen/UpgradeToGeneratedRegexCodeFixer.cs

```diff
@@ -282,25 +282,47 @@ static RegexOptions GetRegexOptionsFromArgument(ImmutableArray<IArgumentOperatio
                 }
 
                 Debug.Assert(parameterName is UpgradeToGeneratedRegexAnalyzer.OptionsArgumentName or UpgradeToGeneratedRegexAnalyzer.PatternArgumentName);
-                if (parameterName == UpgradeToGeneratedRegexAnalyzer.OptionsArgumentName)
+
+                // Check if it's a constant field reference (class-level) - these should be preserved
+                if (argument.Value is IFieldReferenceOperation fieldReferenceOperation &&
+                    fieldReferenceOperation.Member is IFieldSymbol fieldSymbol && fieldSymbol.IsConst)
                 {
-                    string optionsLiteral = Literal(((RegexOptions)(int)argument.Value.ConstantValue.Value!).ToString());
-                    return SyntaxFactory.ParseExpression(optionsLiteral);
+                    return argument.Value.Syntax;
+                }
+
+                // Local constants should be expanded because they won't be in scope for the generated method
+                if (argument.Value is ILocalReferenceOperation localReferenceOperation &&
+                    localReferenceOperation.Local is ILocalSymbol localSymbol && localSymbol.IsConst)
+                {
+                    // For local constants, expand to the value instead of preserving the name
+                    if (parameterName == UpgradeToGeneratedRegexAnalyzer.OptionsArgumentName)
+                    {
+                        string optionsLiteral = Literal(((RegexOptions)(int)argument.Value.ConstantValue.Value!).ToString());
+                        return SyntaxFactory.ParseExpression(optionsLiteral);
+                    }
+                    // For patterns, let the normal logic below handle the formatting
                 }
-                else if (argument.Value is ILiteralOperation literalOperation)
+
+                // Handle literal operations for pattern
+                if (parameterName == UpgradeToGeneratedRegexAnalyzer.PatternArgumentName && argument.Value is ILiteralOperation literalOperation)
```

Comment:
Why is this only relevant to the pattern argument and not to the option argument?